public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx

```diff
@@ -121,10 +123,12 @@ const getStyles = stylesFactory((theme: GrafanaTheme, height: number) => {
 export function RichHistoryQueriesTab(props: Props) {
   const {
     queries,
+    totalQueries,
     loading,
     richHistorySearchFilters,
     updateFilters,
     clearRichHistoryResults,
+    loadMoreRichHistory,
     richHistorySettings,
     exploreId,
     height,
```

Comment:
Not with this line but in this file, line 238 where it displays the number of queries per heading. This is now showing the page size, not the total number of queries for the date. I think it would make more sense if this showed the total number of queries instead of the page size. The user would only run into the pagination if it was relevant then - when they scrolled to the bottom.

Reply:
> This is now showing the page size, not the total number of queries for the date. 

Yeah good point. It shows number of entries for given date shown on the current page (which may be confusing if there's more on the next page).

<img width="254" alt="Screen Shot 2022-05-21 at 17 25 03" src="https://user-images.githubusercontent.com/745532/169664791-3b446a3b-bebe-4db1-a55c-162e0d0a7c77.png">

I'm keen to remove it altogether - I think it might be a bit cumbersome to implement it in the backend.

> I think it would make more sense if this showed the total number of queries instead of the page size. 

Would it make sense to show it at the top and at the bottom?

Reply:
Hiding number of queries per date when not all results are loaded added in 3b76a7bd78963b7f2d166b4488d84dac9a26ac81

Reply:
Alternatively you could change this to say "Displaying x queries" for clarity. 

One thing that I'm not understanding though is it looks like maybe the pagination isn't actually saving us anything in terms of database load (it does save us client rendering load). We aren't doing a COUNT in the sql from what I can tell, we're selecting all queries and then not using any of the data except for the number of results. The PR where this was introduced had a lot of back and forth so I'm not sure if there's a discussion in there I'm missing https://github.com/grafana/grafana/blob/main/pkg/services/queryhistory/database.go#L88

If I'm not, since we already have all the queries in go, we should be able to do that summation by day. If there is a count abstracted in there somewhere, I wonder what the additional load would be to COUNT with a GROUP BY on the day.

Reply:
Good point about COUNT. If we use it we will also get better DB performance. I've added it to to-do list in https://github.com/grafana/grafana/issues/44327 (cc @ivanahuckova). I think could be added in a separate PR.

Reply:
> Alternatively you could change this to say "Displaying x queries" for clarity.

Added in c7ef2b49214d7e91626a7b279684d6331331a764